File: windows/src/GoalMaker.Core/Planning/GoalProgress.cs
namespace GoalMaker.Core.Planning;

/// <summary>Where a goal stands: <paramref name="Value"/> of <paramref name="Target"/>, <paramref name="Fraction"/> (0 to 1) for its ring or bar, and whether it is <paramref name="Hit"/>.</summary>
public sealed record GoalProgress(double Value, double Target, double Fraction, bool Hit);

File: android/app/src/main/kotlin/com/goalmaker/app/application/planning/GoalProgress.kt
package com.goalmaker.app.application.planning

/** Where a goal stands: [value] of [target], [fraction] (0 to 1) for its ring or bar, and whether it is [hit]. */
data class GoalProgress(
    val value: Double,
    val target: Double,
    val fraction: Double,
    val hit: Boolean,
)
